src/app/shared/couchdb.service.ts

```diff
@@ -14,8 +14,8 @@ export class CouchService {
     return Object.assign({}, this.defaultOpts, opts) || this.defaultOpts;
   }
 
-  private couchDBReq(type: string, db: string, opts: any, data?: any) {
-    const url = this.baseUrl + db;
+  private couchDBReq(type: string, db: string, opts: any, data?: any, parentLink?: any) {

```

Comment:
We should include parentLink in the opts variable and adjust the rest of the code to match.  I'd rather we have one optional parameter for the methods available on the CouchService.

Reply:
@paulbert httpRequest is using opts so we used another optional (parentLink). If we are going to include parentLink inside opts then we have to remove parentLink while requesting. so which approach we should follow???